File: test-suite/python/security/hardcoded_secrets_buggy.py
import os
from os import environ, getenv


SECRET_KEY = "django_secret_key_1234567890abcdef"
STRIPE_SECRET_KEY: str = "sk_live_python_1234567890abcdef"

auth_config = {
    "JWT_SECRET": "jwt_secret_from_settings_123456",
    "OAUTH_CLIENT_SECRET": "oauth_client_secret_1234567890",
    "SESSION_SECRET": os.environ.get("SESSION_SECRET", "session_secret_fallback_123456"),
    "WEBHOOK_SECRET": getenv("WEBHOOK_SECRET", "webhook_secret_fallback_123456"),
}

settings = {}
settings["API_KEY"] = "api_key_from_dashboard_123456"
settings["SIGNING_SECRET"] = environ.get("SIGNING_SECRET", "signing_secret_fallback_123456")
settings.setdefault("COOKIE_SECRET", "cookie_secret_fallback_123456")

flask_mapping = dict(SECRET_KEY="flask_secret_key_1234567890")


def jwt_secret() -> str:
    return os.getenv("JWT_SECRET", "fallback_jwt_secret_123456")


class ProviderConfig:
    clientSecret = "provider_client_secret_123456"


apiKey = os.environ.get("MAPS_API_KEY", "maps_api_key_1234567890abcdef")

File: test-suite/golang/security/hardcoded_secrets_buggy.go
package security

import (
	"cmp"
	"os"
)

const jwtSecret = "jwt_secret_from_go_const_1234567890"

var stripeSecretKey = "sk_live_go_1234567890abcdef"

var authConfig = map[string]string{
	"SESSION_SECRET": "session_secret_from_go_map_1234567890",
	"WEBHOOK_SECRET": getenv("WEBHOOK_SECRET", "webhook_secret_fallback_go_123456"),
}

type ProviderConfig struct {
	ClientSecret string
}

func providerConfig() ProviderConfig {
	return ProviderConfig{
		ClientSecret: "oauth_client_secret_go_1234567890",
	}
}

func accessToken() string {
	return envOrDefault("ACCESS_TOKEN", "access_token_fallback_go_1234567890")
}

func jwtFallback() string {
	return cmp.Or(os.Getenv("JWT_SECRET"), "fallback_jwt_secret_go_123456")
}

func settings() map[string]string {
	settings := map[string]string{}
	settings["API_KEY"] = "api_key_from_go_dashboard_123456"
	return settings
}

func getenv(key, fallback string) string {
	if value := os.Getenv(key); value != "" {
		return value
	}
	return fallback
}

func envOrDefault(key, fallback string) string {
	return getenv(key, fallback)
}
